apprise/plugins/NotifyEmail.py

```diff
@@ -778,58 +786,79 @@ def send(self, body, title='', notify_type=NotifyType.INFO, attach=None,
             if reply_to:
                 base['Reply-To'] = ','.join(reply_to)
 
-            # bind the socket variable to the current namespace
-            socket = None
+            message = EmailMessage(
+                recipient=to_addr,
+                to_addrs=[to_addr] + list(cc) + list(bcc),
+                body=base.as_string())
+            messages.append(message)
 
-            # Always call throttle before any remote server i/o is made
-            self.throttle()
+        return self.submit(messages)
 
-            try:
-                self.logger.debug('Connecting to remote SMTP server...')
-                socket_func = smtplib.SMTP
-                if self.secure and self.secure_mode == SecureMailMode.SSL:
-                    self.logger.debug('Securing connection with SSL...')
-                    socket_func = smtplib.SMTP_SSL
+    def submit(self, messages: t.List[EmailMessage]):
 
-                socket = socket_func(
-                    self.smtp_host,
-                    self.port,
-                    None,
-                    timeout=self.socket_connect_timeout,
-                )
+        # error tracking (used for function return)
+        has_error = False
 
-                if self.secure and self.secure_mode == SecureMailMode.STARTTLS:
-                    # Handle Secure Connections
-                    self.logger.debug('Securing connection with STARTTLS...')
-                    socket.starttls()
+        # bind the socket variable to the current namespace
+        socket = None
 
-                if self.user and self.password:
-                    # Apply Login credetials
-                    self.logger.debug('Applying user credentials...')
-                    socket.login(self.user, self.password)
+        # Always call throttle before any remote server i/o is made
+        self.throttle()
```

Comment:
The throttling is right here.

Reply:
My concern is say a connection to Google is made.  Previously (the non-optimized version that you have here).  A connection is made... a batch is delivered and subsequent batches are throttled between connections.  The new approach is fine if it's your own Mail server, but is there risks in sending say 40 individual emails all consecutively? Not sure how 3rd party vendors behave when their clients are perceived to be abusing their mail server and/or if they have throttle limits. 

In my mind the throttle() would go above the `socket.sendmail()` call.  Under the hood, `throttle()` does NOT throttle on it's first call.  It merely just captures a time to start deciding if subsequent calls require a throttle or not.

Curious what your thoughts were here?

Reply:
> The new approach is fine if it's your own Mail server, but is there risks in sending say 40 individual emails all consecutively? Not sure how 3rd party vendors behave when their clients are perceived to be abusing their mail server and/or if they have throttle limits.

I think the opposite would be the case. It is a very regular use case. Imagine you would be off-grid for a while, being busy writing lots of emails. They would be stored within your MUA's outbox, and when connecting to the internet, your MUA would send them out, batched over a single connection. There will only be one connection to the SMTP server, and it will get used to transfer all emails that have been queued up.

On the other hand, there is a very high chance of getting flagged when setting up and tearing down a separate SMTP connection per email message, because this is the common behavior of "dumb" spam bots.